tests/test_core.py

```diff
@@ -425,6 +425,61 @@ def __test_bad_tuple_list(tuple_list):
     __test_bad_tuple_list(badargs)
 
 
+def test_ensure_satisfiable_source_time_tuple():
+    # Documenting the expected behavior of _ensure_valid_source_time_tuples
```

Comment:
_ensure_satisfiable_source_time_tuple (not valid)

Reply:
Yep! Fixed. Good catch.